llvm/include/llvm/IR/IRBuilder.h

```diff
@@ -357,6 +357,9 @@ class IRBuilderBase {
 
   void setConstrainedFPCallAttr(CallBase *I) {
     I->addFnAttr(Attribute::StrictFP);
+    MemoryEffects ME = MemoryEffects::inaccessibleMemOnly();
```

Comment:
I think all of the constrained intrinsics currently use inaccessibleMemOnly, but it might be a good idea to get this attribute from the intrinsic itself. If we added an intrinsic for something like sincos, it would access argument memory too.

We should consider adding separate memory effects for fp control and status registers. Obviously that would be a separate change set.

Reply:
It shouldn't be necessary to touch the attributes. The set of intrinsic attributes are fixed (callsite attributes are another thing, but generally should be droppable here)

Reply:
From the LangRef "Calls and invokes with operand bundles have unknown read / write effect on the heap on entry and exit (even if the call target specifies a memory attribute), unless they’re overridden with callsite specific attributes." I assumed that's why the callsite attribute is being set here.

Reply:
If a constrained intrinsic like `experimental_constrained_sqrt` is used, its memory effect is taken from the intrinsic properties. If `sqrt` is used with bundle operands, the memory effect must be set explicitly.